### DISCUSSION TITLE
Why does my code may crash under Linux whenever I want to handle png files.

### DISCUSSION THREAD

**Discussion by feldti** [ImageMagick/ImageMagick#6133] [Development]
### ImageMagick version

ImageMagick 6.9.11-60 Q16 x86_64 2021-01-25, libMagickWand-6.Q16.so.6

### Operating system, version and so on

Ubuntu 22.04/64Bit 

### Description

I'm in the progress of writing a wrapper for Smalltalk against the MagickWand C-API. The problem is, that my first small example crashes, whenever I want to do things with png. Using the command line apps all stuff is working

Here ist my simple code (even if its Smalltalk - one should be able to understand the logic). as you may see, the logic is simple. The source is the built in logo graphic. And I store it under a specific file - nothing more. The code works with "test.jpg", "test.gif", "test.wpro" --- but with a filename like "test.png" it crashes .... (null pointer exceptions).

I installed and compiled the current version and tested it against libMagickWand-7.Q16HDRI.so and I got the same error ...

Has anyone an idea, what this could be ....

Marten

| wand api |
	
	api := MSKMagickWandAPIWrapper Default.
	
	api magickWandGenesis.
	wand := api newMagickWand.

	(api magickReadImage: wand  path: 'logo:') = MagickFalse
		ifTrue:[ self halt ].
	
	api magickResetIterator: wand. 
	
	[ (api magickNextImage: wand) ~= MagickFalse ] whileTrue: [ 
		api 
                   magickWriteImage: wand 
                   path: '/datadisk/projects/picture-postprocessing/test-pictures/pattern/test.png'.		
	].
	
	api 
		destroyMagickWand: wand ;
		magickWandTerminus.

**Comment by dlemstra** [comment]
This sounds like an issue in your wrapper? Maybe you are doing something that breaks the PNG coder? Without a stacktrace with probably cannot really help you.

**Comment by feldti** [comment]
Well, I assume so, even though there is not much logic there ... pointers everywhere ... and if the lines

	api 
               magickWriteImage: wand 
               path: '/datadisk/projects/picture-postprocessing/test-pictures/pattern/test.jpg'

or

	api 
               magickWriteImage: wand 
               path: '/datadisk/projects/picture-postprocessing/test-pictures/pattern/test.gif'

or

	api 
               magickWriteImage: wand 
               path: '/datadisk/projects/picture-postprocessing/test-pictures/pattern/test.tiff'

are working, but the following lines

	api 
               magickWriteImage: wand 
               path: '/datadisk/projects/picture-postprocessing/test-pictures/pattern/test.png'

	api 
               magickWriteImage: wand 
               path: '/datadisk/projects/picture-postprocessing/test-pictures/pattern/test.webp'


are NOT working, then I am actually a little bit lost :-((((

**Reply by dlemstra** [reply]
It's probably something inside `MSKMagickWandAPIWrapper`?

**Reply by feldti** [reply]
I am still searching ... at least under Windows the same code works as expected ...

**Reply by feldti** [reply]
So, I changed the call to:

api 
	magickSetImageFilename: wand filename: '/datadisk/projects/picture-postprocessing/test-pictures/pattern/test.png' ;		
	magickWriteImage: wand path: nil .

so I changed the name of the wand to a different name (in a different api-call) and this actually worked, because an empty test.png is created (0 Bytes long) BEFORE the API call crashed ...
